tests/test_list_marathon_service_instances.py

```diff
@@ -0,0 +1,132 @@
+            'fake--service2.fake--instance.sha.config': mock.MagicMock(instances=5),
+        }
+        mock_get_num_at_risk_tasks.return_value = 0
+        assert set(list_marathon_service_instances.get_service_instances_that_need_bouncing(
+            '/fake/soa/dir')) == {'fake_service.fake_instance'}
+
+
+def test_get_service_instances_that_need_bouncing_no_difference():
+    with contextlib.nested(
+        mock.patch('paasta_tools.list_marathon_service_instances.get_desired_marathon_configs'),
+        mock.patch('paasta_tools.list_marathon_service_instances.get_current_apps'),
+        mock.patch('paasta_tools.list_marathon_service_instances.get_num_at_risk_tasks'),
+    ) as (
+        mock_get_desired_marathon_configs,
+        mock_get_current_apps,
+        mock_get_num_at_risk_tasks,
+    ):
+        mock_get_desired_marathon_configs.return_value = {'fake--service.fake--instance.sha.config': {'instances': 5}}
+        mock_get_current_apps.return_value = {'fake--service.fake--instance.sha.config': mock.MagicMock(instances=5)}
+        mock_get_num_at_risk_tasks.return_value = 0
+        assert set(list_marathon_service_instances.get_service_instances_that_need_bouncing('/fake/soa/dir')) == set()
+
+
+def test_get_service_instances_that_need_bouncing_instances_difference():
+    with contextlib.nested(
+        mock.patch('paasta_tools.list_marathon_service_instances.get_desired_marathon_configs'),
+        mock.patch('paasta_tools.list_marathon_service_instances.get_current_apps'),
+        mock.patch('paasta_tools.list_marathon_service_instances.get_num_at_risk_tasks'),
+    ) as (
+        mock_get_desired_marathon_configs,
+        mock_get_current_apps,
+        mock_get_num_at_risk_tasks,
+    ):
+        mock_get_desired_marathon_configs.return_value = {'fake--service.fake--instance.sha.config': {'instances': 5}}
+        mock_get_current_apps.return_value = {'fake--service.fake--instance.sha.config': mock.MagicMock(instances=4)}
+        mock_get_num_at_risk_tasks.return_value = 0
+        assert set(list_marathon_service_instances.get_service_instances_that_need_bouncing(
+            '/fake/soa/dir')) == {'fake_service.fake_instance'}
+
+
+def test_get_service_instances_that_need_bouncing_at_risk():
+    with contextlib.nested(
+        mock.patch('paasta_tools.list_marathon_service_instances.get_desired_marathon_configs'),
+        mock.patch('paasta_tools.list_marathon_service_instances.get_current_apps'),
+        mock.patch('paasta_tools.list_marathon_service_instances.get_num_at_risk_tasks'),
+    ) as (
+        mock_get_desired_marathon_configs,
+        mock_get_current_apps,
+        mock_get_num_at_risk_tasks,
+    ):
+        mock_get_desired_marathon_configs.return_value = {'fake--service.fake--instance.sha.config': {'instances': 5}}
+        mock_get_current_apps.return_value = {'fake--service.fake--instance.sha.config': mock.MagicMock(instances=5)}
```

Comment:
Can you add a test specifically for when there is one desired app, but 2 different current apps returned?

I don't see such a test and could imagine that if we didn't get the symmetric difference logic correct, it might assume it doesn't need bouncing because the desired app is already in the set?